src/main/java/be/ugent/rml/MappingFactory.java

```diff
@@ -266,11 +266,11 @@ private void parseObjectMapWithCallback(Term objectmap, BiConsumer<MappingInfo,
                         Map<String, Object[]> parameters = new HashMap<>();
 
                         boolean ignoreDoubleQuotesInParent = this.areDoubleQuotesIgnored(store, parentTriplesMap);
-                        SingleRecordFunctionExecutor parent = new ReferenceExtractor(parents.get(0), ignoreDoubleQuotesInParent);
+                        SingleRecordFunctionExecutor parent = new ReferenceExtractor(parents.get(0).toUpperCase(), ignoreDoubleQuotesInParent);
```

Comment:
I see everywhere this `toUpperCase` be added, is there a specific reason for this?

Reply:
Yeah, as you note below, the old code did case-insensitive comparisons on each data lookup (*many* times), new code does it on creation.  Yields around a factor 5-10 in speed on our test data.